packages/common-integration-tests/src/http/http-apis.ts

```diff
@@ -0,0 +1,216 @@
+import {IAuthClient} from '@gomomento/sdk-core/dist/src/clients/IAuthClient';
+import * as https from 'https';
+
+import {v4} from 'uuid';
+
+const SUPER_USER_PERMISSIONS: PermissionScope =
+  new InternalSuperUserPermissions();
+
+function makeRequest(
+  method: string,
+  hostname: string,
+  path: string,
+  headers: Record<string, string>,
+  body?: string
+): Promise<{
+  statusCode: number;
+  statusMessage: string | undefined;
+  body: string;
+}> {
+  return new Promise((resolve, reject) => {
+    const options = {
+      hostname,
+      path,
+      method,
+      headers,
+    };
+
+    const req = https.request(options, res => {
+      let data = '';
+
+      res.on('data', chunk => {
+        data += chunk;
+      });
+
+      res.on('end', () => {
+        resolve({
+          statusCode: res.statusCode || 0,
+          statusMessage: res.statusMessage,
+          body: data,
+        });
+      });
+    });
+
+    req.on('error', reject);
+
+    if (body) {
+      req.write(body);
+    }
+
+    req.end();
+  });
+}
+
+function putValue(
+  baseUrl: string,
+  apiKey: string,
+  cacheName: string,
+  key: string,
+  value: string,
+  ttl: number
+): Promise<{statusCode: number; statusMessage: string | undefined}> {
+  const path = `/cache/${encodeValue(cacheName)}?key=${encodeValue(
+    key
+  )}&value=${encodeValue(value)}&ttl_seconds=${ttl}`;
+  const headers = {
+    Authorization: apiKey,
+    'Content-Type': 'application/json',
+  };
+
+  return makeRequest('PUT', baseUrl, path, headers, value);
+}
+
+function getValue(
+  baseUrl: string,
+  apiKey: string,
+  cacheName: string,
+  key: string
+): Promise<{
+  statusCode: number;
+  statusMessage: string | undefined;
+  body: string;
+}> {
+  const path = `/cache/${encodeValue(cacheName)}?key=${encodeValue(key)}`;
+  const headers = {
+    Authorization: apiKey,
+  };
+
+  return makeRequest('GET', baseUrl, path, headers);
+}
+
+function deleteValue(
+  baseUrl: string,
+  apiKey: string,
+  cacheName: string,
+  key: string
+): Promise<{statusCode: number; statusMessage: string | undefined}> {
+  const path = `/cache/${encodeValue(cacheName)}?key=${encodeValue(key)}`;
+  const headers = {
+    Authorization: apiKey,
+  };
+
+  return makeRequest('DELETE', baseUrl, path, headers);
+}
+
+function encodeValue(value: string): string {
+  return encodeURIComponent(value);
+}
+
+export function runHttpApiTest(
+  sessionTokenAuthClient: IAuthClient,
+  cacheName: string
+) {
+  describe('Momento HTTP API', () => {
+    let apiKey: string;
+    let baseUrl: string;
+
+    beforeAll(async () => {
+      const resp = await sessionTokenAuthClient.generateApiKey(
+        SUPER_USER_PERMISSIONS,
+        ExpiresIn.minutes(5)
+      );
+      expect(resp).toBeInstanceOf(GenerateApiKey.Success);
+      const successResp = resp as GenerateApiKey.Success;
+      apiKey = successResp.apiKey;
+      baseUrl = `api.cache.${successResp.endpoint}`;
```

Comment:
I might be overlooking something, but would it be possible to call `credentialProvider.getCacheEndpoint()` or similar here in order to construct the endpoint instead of making a new API key each time?

Reply:
it's in the beforeAll hook, so should be called once?